contracts/LimitsChecker.sol

```diff
@@ -0,0 +1,289 @@
+        limit = uint128(_limit);
+
+        /// set spent to _limit if it's greater to avoid math underflow error
+        if (spentAmount > _limit) {
+            spentAmount = uint128(_limit);
+        }
+
+        emit LimitsParametersChanged(_limit, _periodDurationMonths);
+    }
+
+    /// @notice Returns limit and periodDurationMonths
+    /// @return limit - the maximum that can be spent in a period
+    /// @return periodDurationMonths - length of period in months
+    function getLimitParameters() external view returns (uint256, uint256) {
+        return (limit, periodDurationMonths);
+    }
+
+    /// @notice Returns state of the current period: amount spent, balance available for spending,
+    /// @notice start date of the current period and end date of the current period
+    /// @notice If period advanced and the period was not shifted,
+    /// @notice then the method will return spendable balance corresponding to the previous period.
+    /// @return _alreadySpentAmount - amount already spent in the current period
+    /// @return _spendableBalanceInPeriod - balance available for spending in the current period
+    /// @return _periodStartTimestamp - start date of the current period
+    /// @return _periodEndTimestamp - end date of the current period
+    function getPeriodState()
+        external
+        view
+        returns (
+            uint256 _alreadySpentAmount,
+            uint256 _spendableBalanceInPeriod,
+            uint256 _periodStartTimestamp,
+            uint256 _periodEndTimestamp
+        )
+    {
+        return _getCurrentPeriodState();
+    }
+
+    // ------------------
+    // PRIVATE METHODS
+    // ------------------
+    function _getCurrentPeriodState()
+        internal
+        view
+        returns (
+            uint256 _alreadySpentAmount,
+            uint256 _spendableBalanceInPeriod,
+            uint256 _periodStartTimestamp,
+            uint256 _periodEndTimestamp
+        )
+    {
+        return (
+            spentAmount,
+            _spendableBalance(),
+            _getPeriodStartFromTimestamp(currentPeriodEndTimestamp - 1),
+            currentPeriodEndTimestamp
+        );
+    }
+
+    function _spendableBalance() internal view returns (uint256) {
+        return limit - spentAmount;
+    }
+
+    function _validatePeriodDurationMonths(uint256 _periodDurationMonths) internal view {
+        require(
+            _periodDurationMonths == 1 ||
+                _periodDurationMonths == 2 ||
+                _periodDurationMonths == 3 ||
+                _periodDurationMonths == 6 ||
+                _periodDurationMonths == 12,
+            ERROR_INVALID_PERIOD_DURATION
+        );
+    }
+
+    function _getPeriodStartFromTimestamp(uint256 _timestamp) internal view returns (uint256) {
+        // Get year and number of month of the timestamp:
+        (uint256 _year, uint256 _month, ) = bokkyPooBahsDateTimeContract.timestampToDate(
```

Comment:
please, consider removing underscores throughout the method body for locals names